mbuild/tests/test_utils.py

```diff
@@ -186,3 +188,182 @@ def test_has_ipython(self):
 
     def test_removed_func_error(self):
         RemovedFuncError("old_function", "new_function", "0.0.0", "0.1.0")
+
+
+class TestUtilsConversion(BaseTest):
+    def test_RB_to_OPLS_c5_not_0(self):
+        with pytest.raises(
+            ValueError,
+            match=r"c5 must equal zero, so this conversion is not possible.",
+        ):
+            c0 = 0.1
+            c1 = 0.1
+            c2 = -0.2
+            c3 = -0.1
+            c4 = -0.2
+            c5 = 0.3
+            RB_to_OPLS(c0, c1, c2, c3, c4, c5)
+
+    def test_RB_to_OPLS_f0_not_0_within_tolerance_warn(self):
+        # should throw a warning that f0 is not zero
+        text_for_error_tol = (
+            "f0 = 2 * (c0 + c1 + c2 + c3 + c4 + c5) is not zero. "
+            "The f0/2 term is the constant for the OPLS dihedral. "
+            "Since the f0 term is not zero, the dihedral is not an "
+            "exact conversion; since this constant does not contribute "
+            "to the force equation, this should provide matching results "
+            "for MD, but the energy for each dihedral will be shifted "
+            "by the f0/2 value."
+        )
+        with pytest.warns(UserWarning):
+            warn(f"{text_for_error_tol}", UserWarning)
```

Comment:
I think by having line 219 raise a warning we make this test pass artificially instead of testing that the function on line 226 raises a warning.

Reply:
This is fixed